queries/reflect.go

```diff
@@ -303,10 +487,66 @@ Rows:
 	}
 
 	if bkind == kindStruct && !foundOne {
-		return sql.ErrNoRows
+		return nil, sql.ErrNoRows
 	}
 
-	return nil
+	return findNullTables(cols, nullCols), nil
+}
+
+// findNullTables: if every value for a given table was nil, then add it to our list of null tables
+// this is used to set table.R.ForeignTable = nil instead of having it populate with zero values
+// both string slices passed to this function will contain entries like "table.column"
+func findNullTables(allColumns, nullColumns []string) []string {
+	nullTables := make([]string, 0)
+
+	allColumnsMap := tableColumnMap(allColumns)
+	nullColumnsMap := tableColumnMap(nullColumns)
+
+	for table, columns := range nullColumnsMap {
+		// if all the columns in the null map are the same as all the columns in the table, then it's a null table
+		if len(columns) == len(allColumnsMap[table]) {
```

Comment:
Might be nicer to do an early `continue` here by inverting the condition

Reply:
agreed!